src/app/services/bucket/qarnot.py

```diff
@@ -0,0 +1,84 @@
+import os
+import logging
+from typing import List, Optional
+from qarnot.connection import Connection
+from qarnot.bucket import Bucket
+
+from app import config as cfg
+
+
+__all__ = ['QarnotBucket']
+
+
+logger = logging.getLogger("uvicorn.warning")
+
+
+class QarnotBucket:
+
+    _bucket: Optional[Bucket] = None
+    _media_folder: Optional[str] = None
```

Comment:
It was more for a clearer code layout since the constructor initializes the value, but it calls a function that needs to check if the values are none.
Their default values are shared across all instances, but each instance can have different values for this attribute since they are set in the constructor. Here is an example:
```
In [1]: class A:
   ...:     a = None
   ...:     def __init__(self, val, ref=2):
   ...:         if val > ref:
   ...:             self.a = val
   ...: 

In [2]: a = A(0)

In [3]: b = A(3)

In [4]: c = A(4)

In [5]: print(a.a, b.a, c.a)
None 3 4
```

Reply:
Yes you're right (the fact of sharing variable might become dangerous when it concern mutable types)